From dca21fc335bcb49aa6d6134cf8f33716221d6ced Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Wed, 2 Sep 2020 09:38:59 +0800
Subject: [PATCH] Revert "power/avs: rockchip-io-domain: rv1126 support to
 control vccio1"

This reverts commit df8465ab9d5366f5ae65ae109116653ce278a980.

Change-Id: I62eb3f42bd4e4ada3ec16471ca8a9c4f5c3efc9d
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 drivers/power/avs/rockchip-io-domain.c | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/drivers/power/avs/rockchip-io-domain.c b/drivers/power/avs/rockchip-io-domain.c
index b6e2ed384582..ce902fbb6493 100644
--- a/drivers/power/avs/rockchip-io-domain.c
+++ b/drivers/power/avs/rockchip-io-domain.c
@@ -43,10 +43,6 @@
 #define PX30_IO_VSEL_VCCIO6_SRC		BIT(0)
 #define PX30_IO_VSEL_VCCIO6_SUPPLY_NUM	1
 
-#define RV1126_PMUGRF_IO_VSEL			0x140
-#define RV1126_PMUGRF_IO_VSEL_VCCIO1_SRC	BIT(0)
-#define RV1126_PMUGRF_IO_VSEL_VCCIO1_SUPPLY_NUM	1
-
 #define RK3288_SOC_CON2			0x24c
 #define RK3288_SOC_CON2_FLASH0		BIT(7)
 #define RK3288_SOC_FLASH_SUPPLY_NUM	2
@@ -182,26 +178,6 @@ static void px30_iodomain_init(struct rockchip_iodomain *iod)
 		dev_warn(iod->dev, "couldn't update vccio0 ctrl\n");
 }
 
-static void rv1126_pmu_iodomain_init(struct rockchip_iodomain *iod)
-{
-	int ret;
-	u32 val;
-
-	/* if no pmu io supply we should leave things alone */
-	if (!iod->supplies[RV1126_PMUGRF_IO_VSEL_VCCIO1_SUPPLY_NUM].reg)
-		return;
-
-	/*
-	 * set pmu io iodomain to also use this framework
-	 * instead of a special gpio.
-	 */
-	val = RV1126_PMUGRF_IO_VSEL_VCCIO1_SRC |
-	      (RV1126_PMUGRF_IO_VSEL_VCCIO1_SRC << 16);
-	ret = regmap_write(iod->grf, RV1126_PMUGRF_IO_VSEL, val);
-	if (ret < 0)
-		dev_warn(iod->dev, "couldn't update pmu io iodomain ctrl\n");
-}
-
 static void rk3288_iodomain_init(struct rockchip_iodomain *iod)
 {
 	int ret;
@@ -513,7 +489,6 @@ static const struct rockchip_iodomain_soc_data soc_data_rv1126_pmu = {
 		"pmuio0",
 		"pmuio1",
 	},
-	.init = rv1126_pmu_iodomain_init,
 };
 
 static const struct of_device_id rockchip_iodomain_match[] = {
-- 
2.35.3

